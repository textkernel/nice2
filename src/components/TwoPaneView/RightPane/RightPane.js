import React from 'react';
import PropTypes from 'prop-types';
import bem from 'bem';
import styles from './RightPane.scss';

const { block } = bem({
    name: 'RightPane',
    classnames: styles,
});

const RightPane = props => {
    const { children, ...rest } = props;

    return (
        <div {...rest} {...block(props)}>
            {children}
        </div>
    );
};

RightPane.displayName = 'RightPane';

RightPane.propTypes = {
    /** Node(s) to be rendered in the right pane */
    children: PropTypes.node.isRequired,
};

export default RightPane;
