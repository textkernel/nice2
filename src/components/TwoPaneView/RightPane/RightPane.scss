.RightPane {
    position: absolute;
    left: calc(var(--two-pane-left-width) + var(--site-container-padding));
    display: inline-block;
    box-sizing: border-box;
    overflow: hidden;
}
