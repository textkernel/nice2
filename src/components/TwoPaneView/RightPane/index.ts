export { default as RightPane } from './RightPane';
export { Props as RightPaneProps } from './RightPane';
