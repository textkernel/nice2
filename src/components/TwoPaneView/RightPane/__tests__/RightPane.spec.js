import React from 'react';
import toJson from 'enzyme-to-json';
import { RightPane } from '../RightPane';

describe('<RightPane> that renders a right pane of a two pane view', () => {
    it('should render correctly', () => {
        const wrapper = shallow(<RightPane>Some content</RightPane>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
