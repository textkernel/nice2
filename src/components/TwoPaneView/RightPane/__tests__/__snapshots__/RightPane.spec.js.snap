// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RightPane> that renders a right pane of a two pane view should render correctly 1`] = `
<div
  className="RightPane"
>
  Some content
</div>
`;
